org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -16,63 +16,99 @@ namespace MixedReality.Toolkit.Input
     [AddComponentMenu("MRTK/Input/Near Interaction Mode Detector")]
     public class NearInteractionModeDetector : ProximityDetector
     {
+        /// <summary>
+        /// The set of near interactables that belongs to near interaction
+        /// </summary>
         [SerializeField]
-        [Tooltip("The set of near interactors that belongs to near interaction")]
-        private List<XRBaseInteractor> nearInteractors;
+        [Tooltip("The set of near interactables that belongs to near interaction")]
+        private List<XRBaseInteractable> nearInteractables;
 
         /// <summary>
-        /// Used to keep track of the previously detected colliders so that we can know which
-        /// colliders stopped being detected and update their buttons front plate and rounded
-        /// rect if they are labeled as dynamic (based on proximity).
+        /// Used to keep track of the previously detected interactrables so that we can know which
+        /// interactable stopped being detected and update their buttons front plate RawImage.
         /// </summary>
-        List<Collider> previouslyDetectedColliders = new List<Collider>();
+        private HashSet<IXRProximityInteractable> previouslyDetectedColliderInteractableDuples = new();
 
         /// <inheritdoc />
         public override bool IsModeDetected()
         {
             bool result = base.IsModeDetected() || IsNearInteractorSelecting();
             if (result)
             {
-                for (int i = 0; i < previouslyDetectedColliders.Count; i++)
+                UpdateProximityExited();
+                UpdateProximityEntered();
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Calls OnProximityExited for all interactables that were previously ProximityHover detected but are no longer detected.
+        /// </summary>
+        private void UpdateProximityExited()
+        {
+            List<IXRProximityInteractable> currentlyDetectedInteractable = GetCurrentlyDetectedColliderInteractablesDuples();
```

Comment:
You're calling GetCurrentlyDetectedColliderInteractablesDuples() twice per frame.  Call once, and share the result with UpdateProximityExited() and UpdateProximityEntered()

Reply:
Done in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/659/commits/7ee0fe33c7946426a3a643e4c266e4defd1fc3e2 resolving